org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/sso/filter/SSOWebuiFilter.java

```diff
@@ -0,0 +1,133 @@
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.           *
+ * See the GNU General Public License for more details.                       *
+ * You should have received a copy of the GNU General Public License along    *
+ * with this program; if not, write to the Free Software Foundation, Inc.,    *
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.                     *
+ *****************************************************************************/
+package org.adempiere.webui.sso.filter;
+
+import java.io.IOException;
+import java.util.logging.Level;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.adempiere.base.sso.ISSOPrinciple;
+import org.adempiere.base.sso.SSOUtils;
+import org.compiere.model.MSysConfig;
+import org.compiere.util.CLogger;
+
+/**
+ * Request filter class for the SSO authentication
+ * 
+ * @author Logilite Technologies
+ */
+public class SSOWebuiFilter implements Filter
+{
+	/** Logger */
+	protected static CLogger		log				= CLogger.getCLogger(SSOWebuiFilter.class);
+	// TODO as this is static, will not work on multi tenant environment
+	private static ISSOPrinciple	m_SSOPrinciple	= null;
+
+	/**
+	 * AdempiereMonitorFilter
+	 */
+	public SSOWebuiFilter()
+	{
+		super();
+	} // AdempiereMonitorFilter
+
+	/**
+	 * Filter
+	 * 
+	 * @param  request          request
+	 * @param  response         response
+	 * @param  chain            chain
+	 * @throws IOException
+	 * @throws ServletException
+	 */
+	public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException
+	{
+		boolean isSSOEnable = MSysConfig.getBooleanValue(MSysConfig.ENABLE_SSO, false);
+		if (isSSOEnable && request instanceof HttpServletRequest)
+		{
+			HttpServletRequest httpRequest = (HttpServletRequest) request;
+			HttpServletResponse httpResponse = (HttpServletResponse) response;
+			boolean isRedirectToLoginOnError = false; 
+			try
+			{
+				if (m_SSOPrinciple == null)
+				{
+					m_SSOPrinciple = SSOUtils.getSSOPrinciple();
+				}
+
+				if (m_SSOPrinciple != null)
+				{
+					if (m_SSOPrinciple.hasAuthenticationCode(httpRequest, httpResponse))
+					{
+						// Use authentication code get get token
+						m_SSOPrinciple.getAuthenticationToken(httpRequest, httpResponse, SSOUtils.SSO_MODE_WEBUI);
+					}
+					else if (!m_SSOPrinciple.isAuthenticated(httpRequest, httpResponse))
+					{
+						// Redirect to SSO sing in page for authentication
+						m_SSOPrinciple.redirectForAuthentication(httpRequest, httpResponse, SSOUtils.SSO_MODE_WEBUI);
+						return;
+					}
+					else if (m_SSOPrinciple.isAccessTokenExpired(httpRequest, httpResponse))
```

Comment:
access token is not needed after the initial authentication, this expire and refresh check is unnecessary.

Reply:
Removed this check